packages/server/src/api/rest/index.ts

```diff
@@ -220,29 +237,60 @@ class RequestHandler extends APIHandlerBase {
     // divider used to separate compound ID fields
     private idDivider;
 
-    private urlPatterns;
+    private urlPatternMap: Record<UrlPatterns, UrlPattern>;
+    private modelNameMapping: Record<string, string>;
+    private reverseModelNameMapping: Record<string, string>;
+    private prefix: string | undefined;
 
     constructor(private readonly options: Options) {
         super();
         this.idDivider = options.idDivider ?? prismaIdDivider;
         const segmentCharset = options.urlSegmentCharset ?? 'a-zA-Z0-9-_~ %';
-        this.urlPatterns = this.buildUrlPatterns(this.idDivider, segmentCharset);
+
+        this.prefix = options.prefix;
+        this.modelNameMapping = options.modelNameMapping ?? {};
+        this.reverseModelNameMapping = Object.fromEntries(
+            Object.entries(this.modelNameMapping).map(([k, v]) => [v, k])
+        );
+        this.urlPatternMap = this.buildUrlPatternMap(segmentCharset);
     }
 
-    buildUrlPatterns(idDivider: string, urlSegmentNameCharset: string) {
+    private buildUrlPatternMap(urlSegmentNameCharset: string): Record<UrlPatterns, UrlPattern> {
         const options = { segmentValueCharset: urlSegmentNameCharset };
+
+        const buildPath = (segments: string[]) => {
```

Comment:
if you prefer we can also drop this change and use `string` rather than `string[]` in the constructor of `UrlPattern`.